redbot/cogs/mod/kickban.py

```diff
@@ -214,6 +225,19 @@ class KickBanMixin(MixinMeta):
         If days is not a number, it's treated as the first word of the reason.
         Minimum 0 days, maximum 7. Defaults to 0."""
 
+        author = ctx.author
+        guild = ctx.guild
+        if reason is None:
+            reason = _("No reason was given.")
+        toggle = await self.settings.guild(guild).toggle_dm()
+        if toggle:
+            if guild.me.top_role > user.top_role and user != guild.owner and author != user:
+                with contextlib.suppress(discord.HTTPException):
+                    em = discord.Embed(
+                        title=_(bold("You have been banned from {guild}.").format(guild=guild))
+                    )
+                    em.add_field(name=_("**Reason**"), value=reason, inline=False)
+                    await user.send(embed=em)
```

Comment:
You should move this section to `ban_user()` method (after the `if` checks but before the ban) so that you don't have to check for stuff that is already checked there.